miner/worker.go

```diff
@@ -161,10 +161,6 @@ func (w *worker) commitNewWork(predicateContext *precompileconfig.PredicateConte
 		gasLimit = core.CalcGasLimit(parent.GasUsed, parent.GasLimit, ap1.GasLimit, ap1.GasLimit)
 	}
 
-	extra, err := header.ExtraPrefix(w.chainConfig, parent, timestamp)
```

Comment:
Should we also move this since it appends to the extra in header? Not sure why we need to keep it in two different places. Moving this will probably require to pass the env or predicateResults though
https://github.com/ava-labs/coreth/blob/move-predicate-header-parsing/miner/worker.go#L470-L476

Reply:
If we are okay with changing the signature of `FinalizeAndAssemble` then we can do that

Reply:
I guess we don't have many options. Otherwise it seems we need to create the extra here with gap to be later filled by `FinalizeAndAssemble`

Reply:
uh I misread that,. we actually prepend in the `FinalizeAndAssemble` it should be fine.